src/nccl_ofi_rdma.c

```diff
@@ -2955,7 +2955,13 @@ static int reg_mr_ep(nccl_net_ofi_rdma_ep_t *ep,
 		/* Cache miss */
 	}
 
-	ret = reg_mr_on_device(ep, ckey, type, &ret_handle);
+	if (mr_cache) {
```

Comment:
Looking more, there's no reason to do this just in the mr cache case.

Reply:
Move the logic into ckey creation.